packages/node/src/nodeClient.ts

```diff
@@ -9,29 +9,52 @@ export class NodeClient implements Client<Options> {
   /** Options for the client. */
   protected readonly _options: Options;
 
+  private _events: Event[];
+  private _transport: Transport;
+
   /**
    * Initializes this client instance.
    *
    * @param apiKey API key for your project
    * @param options options for the client
    */
-  public constructor(apiKey: string, options: Options) {
+  public constructor(apiKey: string, options: Options = {}) {
     this._apiKey = apiKey;
     this._options = options;
+    this._events = [];
+    this._transport = this._setupTransport();
+
+    const uploadIntervalInSec = options.uploadIntervalInSec ?? 30;
+    var self = this;
+    setInterval(function() {
```

Comment:
I think this will stop the process from shutting down properly.

You need to call `unref()` on the interval https://nodejs.org/api/timers.html#timers_timeout_unref

Alternatively you could switch from a constant interval to only creating a timeout when an event is actually logged i.e. a debounce rather than a constant loop. That might also help give us the guaranteed log on process shutdown.

Reply:
You mean calling `setTimeout` when logEvent happens? If the timeout is quite long enough, it still will prevent process shutting down for a certain time period, right?

Reply:
> You mean calling setTimeout when logEvent happens?

Yes.

If the timeout is quite long enough, it still will prevent process shutting down for a certain time period, right?

Yes.

Even though this is just a small part of the overall SDK behavior I think it's important to put a lot of thought into this. Customers might not necessarily be able to articulate what the correct behavior should be but I bet if this doesn't feel intuitive they'll hate it. Maybe some customer research could help define what folks are expecting to happen?